Odata-docs/client/using-dataservice-ihttpclientfactory.md

```diff
@@ -0,0 +1,82 @@
+# Using DataserviceContext IHttpClientFactory in Microsoft.OData.Client
+
+Ever wanted to use `HttpClient` with `Microsoft.OData.Client`? Now you can! In this post, we'll show you how to do it easily and highlight the benefits.
+
+## Why Use IHttpClientFactory?
+
+IHttpClientFactory is a feature in .NET Core that helps you manage HttpClient instances. Here are some benefits:
+- **Automatic Management**: It handles the lifecycle of HttpClient instances, preventing issues like socket exhaustion.
+- **Configuration**: You can set up policies like retries and timeouts.
+- **Dependency Injection**: It integrates with .NET Core's DI system, making it easy to use in your services.
+
+## Getting Started
+
+To use **HttpClient** with `Microsoft.OData.Client`, you can use the `HttpClientFactory` property of `DataServiceContext`. This lets you inject your custom HttpClient instance.
+
+### Step 1: Create a Custom HttpClientFactory
+
+First, create a custom HttpClientFactory:
+```cs
+public class CustomHttpClientFactory : IHttpClientFactory
+{
+    private readonly HttpClient _httpClient;
+
+    public CustomHttpClientFactory(HttpClient httpClient)
+    {
+        _httpClient = httpClient;
+    }
+
+    public HttpClient CreateClient(string name)
+    {
+        return _httpClient;
+    }
+}
+```
+
+### Step 2: Create and Configure HttpClient
+
+Next, create an HttpClient instance and set the timeout:
+```cs
+var httpClient = new HttpClient
+{
+    Timeout = TimeSpan.FromSeconds(160)
+};
+
+var httpClientFactory = new CustomHttpClientFactory(httpClient);
+
+var context = new Container(new Uri("https://localhost:7214/odata"))
+{
+    HttpClientFactory = httpClientFactory
+};
+```
+
+### Step 3: Using Dependency Injection
```

Comment:
I think we should start with this approach, since it matches the recommended usage from .NET, then show creating custom HttpClientFactory as an alternative.